src/Shouldly/ShouldlyExtensionMethods/ShouldBeDictionaryTestExtensions.cs

```diff
@@ -6,31 +6,60 @@ namespace Shouldly;
 [ShouldlyMethods]
 public static partial class ShouldBeDictionaryTestExtensions
 {
-    public static void ShouldContainKey<TKey, TValue>(this IDictionary<TKey, TValue> dictionary, TKey key, string? customMessage = null)
+    public static void ShouldContainKey<TKey, TValue>(this IEnumerable<KeyValuePair<TKey, TValue>> dictionary, TKey key, string? customMessage = null)
```

Comment:
this is unfortunately a breaking runtime change. both overloads need to exist for this to be deployed in a minor

Reply:
@SimonCropp I can't see the runtime break. Can you describe it? It might be palatable as long as it doesn't affect mainstream usage of `ShouldContainKey`.

Specifically, `IDictionary<TKey,TValue>` inherits `IEnumerable<KeyValuePair<TKey, TValue>>`. Anything that was passed previously in a method call will behave the same as before.

Reply:
at runtime consumers will bind to `ShouldContainKey(IDictionary)`. so if u have a 3rd party dll that is consuming `ShouldContainKey(IDictionary)`, and that dll is not recompiled, there will be a missing method exception

Reply:
Are you sure the runtime doesn't make things work dynamically at runtime? The reason I ask is that the very first thing in the .NET breaking changes rules says that this is specifically not a breaking change: https://github.com/dotnet/runtime/blob/main/docs/coding-guidelines/breaking-change-rules.md

Reply:
Okay, upon reading the whole first 'Allowed' entry under https://github.com/dotnet/runtime/blob/main/docs/coding-guidelines/breaking-change-rules.md, this would not be allowed. Huh, I wonder what bullets I've dodged 😆